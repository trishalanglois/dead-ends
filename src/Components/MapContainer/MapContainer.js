import React, { Component } from 'react';
import { Map, InfoWindow, Marker, GoogleApiWrapper } from 'google-maps-react';


const mapStyles = {
  width: '80%',
  height: '80%'
}

export class MapContainer extends Component {
  render() {
    return (
      <Map
        google={this.props.google}
        zoom={14}
        style={mapStyles}
        initialCenter={{
          lat: 37.774929,
          lng: -122.419416,
        }}
      />
    );
  }

}

export default GoogleApiWrapper({
  apiKey: 'AIzaSyBFQ1czovRlzvB4l3QxC5YwoxJ_URDKP6g'
})(MapContainer)
