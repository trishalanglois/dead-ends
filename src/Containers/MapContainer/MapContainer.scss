@import '../../scss/variables';

.map-view {
  background-image: url('../../images/background.jpg');
  background-size: cover;
  background-position: center;
  height: 88vh;
  width: 100vw;
}

.map-container-subheader {
  display: flex;
  justify-content: space-between;
  background-color: $dark-color;
  height: 10%;
  margin-bottom: 0px;
  padding-bottom: 0px;
}

.map-subheader {
  font-family: $accent-font;
  font-size: 2em;
  margin: 0px 1em;
  color: $light-grey;
  font-weight: 200;
  width: auto;
}

.map-card-view {
  float: right;
  margin: 10% 15%;
  position: relative;
  z-index: 100;
}

.card-subtitle {
  font-size: 1.5em;
  margin: 0em;
}

.map-button {
  float: right;
  font-size: 1.35em;
  font-family: $main-font;
  margin: .25em;
  padding: 0em 1em -1em 1em;
  width: 10em;
  height: 2em;
  border-style: none;
  background-color: $medium-grey;
    &:hover {
      cursor: pointer;
      transform: scale(1.01);
      background-color: $dark-grey;
    }
    &:focus {
      background-color: $wishlist-color;
    }
}
