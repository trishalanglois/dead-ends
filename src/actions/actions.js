export const currentUser = user => ({
  type: 'SET_USER',
  name: user
})

export const setLocation = location => ({
  type: 'SET_LOCATION',
  location
})

export const setSpookyLocations = locations => ({
  type: 'SET_SPOOKY_LOCATIONS',
  locations
})

export const updateFavorites = location => ({
  type: 'UPDATE_FAVORITES',
  location
})

export const updateClickedCard = location => ({
  type: 'UPDATE_CLICKED_CARD',
  location
})
