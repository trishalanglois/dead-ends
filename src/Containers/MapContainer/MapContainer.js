import React, { Component } from 'react';
import { Map, InfoWindow, Marker, GoogleApiWrapper } from 'google-maps-react';
import './MapContainer.scss';
import { connect } from 'react-redux';
import { updateFavorites } from '../../actions/actions';
import { Link } from 'react-router-dom';

const mapStyles = {
  width: '40%',
  height: '50%',
  margin: '5%',
  border: '1px solid black'
}

export class MapContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedPlace: `${this.props.currentMap.location}`
    }
  }

onClose = props => {
 if (this.state.showingInfoWindow) {
   this.setState({
     showingInfoWindow: false,
     activeMarker: null
   });
  }
};

  render() {
    return (
      <main className="map-view">
        <div className="map-container-subheader">
          <Link to={'/locations'}>
            <h2 className="map-subheader">Return to all locations</h2>
          </Link>
          <Link to={'/favorites'}>
            <h3 className="favorites-button">View Favorites</h3>
          </Link>
        </div>
          <Map
            className="map-component"
            google={this.props.google}
            zoom={14}
            style={mapStyles}
            initialCenter={{
              lat: this.props.currentMap.latitude,
              lng: this.props.currentMap.longitude
            }}
          >
            <Marker
                onClick={this.onMarkerClick}
                name= {this.props.currentMap.location}
              />
              <InfoWindow
                marker={this.state.activeMarker}
                visible={this.state.showingInfoWindow}
                onClose={this.onClose}
              >
                <div>
                  <h4>{this.state.selectedPlace.location}</h4>
                </div>
              </InfoWindow>
          </Map>
          <aside className="card map-card-view">
            <div className="card-title-container">
            <h3 className="card-title">{this.props.currentMap.location}</h3>
            <button className="map-button" onClick={() => this.props.updateFavorites(this.props.currentMap)}>Add to Favorites</button>
            </div>
            <h4 className="card-subtitle">{this.props.currentMap.city}, {this.props.currentMap.state}</h4>
            <p className="card-description">{this.props.currentMap.description}</p>
          </aside>
      </main>
    )
  }
}

export const mapStateToProps = state => ({
  currentMap: state.currentMap
})

export const mapDispatchToProps = dispatch => ({
  updateFavorites: newLocation => dispatch(updateFavorites(newLocation)),
})

const wrappedMapContainer = GoogleApiWrapper({
  apiKey: 'AIzaSyBFQ1czovRlzvB4l3QxC5YwoxJ_URDKP6g'
})(MapContainer)

export default connect(mapStateToProps, mapDispatchToProps)(wrappedMapContainer)
