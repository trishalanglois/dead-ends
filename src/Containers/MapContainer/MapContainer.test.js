import React from 'react';
import { MapContainer, mapStateToProps, mapDispatchToProps } from './MapContainer';
import { shallow } from 'enzyme';
import { updateFavorites } from '../../actions/actions';

describe('MapContainer', () => {
  const mockProps = {
    currentMap: {
      "city": "Denver",
      "country": "United States",
      "description": "BOO!",
      "location": "Ada Cemetery",
      "state": "Colorado",
      "state_abbrev": "CO",
      "longitude": -85.50489309999999,
      "latitude": 42.9621061,
      "city_longitude": -85.4954803,
      "city_latitude": 42.960727
    }
  }
  let wrapper = shallow(<MapContainer
    currentMap={mockProps}
    />)

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return an object with the currentMap', () => {
      const mockState = {
        currentMap: {
          "city": "Denver",
          "country": "United States",
          "description": "BOO!",
          "location": "Ada Cemetery",
          "state": "Colorado",
          "state_abbrev": "CO",
          "longitude": -85.50489309999999,
          "latitude": 42.9621061,
          "city_longitude": -85.4954803,
          "city_latitude": 42.960727
        }
      };
      const expected = {
        currentMap: {
          "city": "Denver",
          "country": "United States",
          "description": "BOO!",
          "location": "Ada Cemetery",
          "state": "Colorado",
          "state_abbrev": "CO",
          "longitude": -85.50489309999999,
          "latitude": 42.9621061,
          "city_longitude": -85.4954803,
          "city_latitude": 42.960727
        }
      }
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call mapDispatchToProps with a updateFavorites action when updateFavorites is called from Props', () => {
      const mockDispatchToProps = jest.fn();
      const actionToDispatch = updateFavorites(mockProps);
      const mappedProps = mapDispatchToProps(mockDispatchToProps);
      mappedProps.updateFavorites(mockProps);
      
      expect(mockDispatchToProps).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})
