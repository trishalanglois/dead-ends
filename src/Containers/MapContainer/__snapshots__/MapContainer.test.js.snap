// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer should match the snapshot 1`] = `
<main
  className="map-view"
>
  <div
    className="map-container-subheader"
  >
    <Link
      to="/locations"
    >
      <h2
        className="map-subheader"
      >
        Return to all locations
      </h2>
    </Link>
    <Link
      to="/favorites"
    >
      <h3
        className="favorites-button"
      >
        View Favorites
      </h3>
    </Link>
  </div>
  <Map
    center={Object {}}
    centerAroundCurrentLocation={false}
    className="map-component"
    containerStyle={Object {}}
    initialCenter={
      Object {
        "lat": undefined,
        "lng": undefined,
      }
    }
    style={
      Object {
        "border": "1px solid black",
        "height": "50%",
        "margin": "5%",
        "width": "40%",
      }
    }
    visible={true}
    zoom={14}
  >
    <Marker
      name="Marker"
    />
    <InfoWindow
      onClose={[Function]}
      visible={false}
    >
      <div>
        <h4 />
      </div>
    </InfoWindow>
  </Map>
  <aside
    className="card map-card-view"
  >
    <div
      className="card-title-container"
    >
      <h3
        className="card-title"
      />
      <button
        className="map-button"
        onClick={[Function]}
      >
        Add to Favorites
      </button>
    </div>
    <h4
      className="card-subtitle"
    >
      , 
    </h4>
    <p
      className="card-description"
    />
  </aside>
</main>
`;
